OpenMarket/OpenMarket/View/GridCollectionViewCell.swift

```diff
@@ -0,0 +1,104 @@
+    
+    private var productImage: UIImageView = {
+        let imageView = UIImageView()
+        imageView.contentMode = .scaleAspectFit
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        
+        return imageView
+    }()
+    
+    private var productName: UILabel = {
+        let label = UILabel()
+        label.font = .preferredFont(forTextStyle: .title3)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        
+        return label
+    }()
+    
+    private var price: UILabel = {
+        let label = UILabel()
+        label.numberOfLines = 2
+        label.translatesAutoresizingMaskIntoConstraints = false
+        
+        return label
+    }()
+    
+    private var stock: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        
+        return label
+    }()
+    
+    private let stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .vertical
+        stackView.alignment = .center
+        stackView.distribution = .equalCentering
+        stackView.spacing = 5
+        
+        return stackView
+    }()
+    
+    // MARK: - initializers
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setSubViews()
+        setStackViewConstraints()
+        setBorder()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setSubViews()
+        setStackViewConstraints()
+        setBorder()
+    }
+    
+    // MARK: - functions
+    
+    private func setSubViews() {
+        self.contentView.addSubview(productImage)
+        self.contentView.addSubview(stackView)
+        [productName, price, stock].forEach { stackView.addArrangedSubview($0) }
+    }
+    
+    private func setStackViewConstraints() {
+        NSLayoutConstraint.activate(
+            [productImage.topAnchor.constraint(equalTo: self.contentView.topAnchor, constant: 5),
+             productImage.leadingAnchor.constraint(equalTo: self.contentView.leadingAnchor),
+             productImage.trailingAnchor.constraint(equalTo: self.contentView.trailingAnchor),
+             productImage.heightAnchor.constraint(equalTo: self.contentView.heightAnchor, multiplier: 0.5)])
+        
+        NSLayoutConstraint.activate(
+            [stackView.leadingAnchor.constraint(equalTo: self.contentView.leadingAnchor),
+             stackView.trailingAnchor.constraint(equalTo: self.contentView.trailingAnchor),
+             stackView.bottomAnchor.constraint(equalTo: self.contentView.bottomAnchor, constant: -5)])
+    }
+    
+    private func setBorder() {
+        self.layer.cornerRadius = 10
+        self.layer.borderWidth = 1.5
+        self.layer.borderColor = UIColor.systemGray3.cgColor
+    }
+    
+    func setViewItems(_ item: ProductDetail) {
+        productImage.image = item.setThumbnailImage()
```

Comment:
재사용 이슈가 발생할 수 있습니다.

prepareForReuse로 이를 해결해주세요,

Reply:
gridViewCell의 경우 prepareForReuse를 사용해서 각 컨텐츠에 nil을 할당해주는 방법으로 해결했습니다. 하지만, [prepareForReuse공식문서](https://developer.apple.com/documentation/uikit/uitableviewcell/1623223-prepareforreuse)에 
` To avoid potential performance issues, you should only reset attributes of the cell that are not related to content, for example, alpha, editing, and selection state.` 와 같은 문구가 있는걸 확인했습니다. 
저희가 nil을 할당해주는 동작도 컨텐츠를 조작한다고 생각이 들었는데, 단순히 nil을 할당해주는 동작은 성능이슈가 없을지 궁금합니다.

또한, list view 같은 경우는 UIListContentConfiguration의 기본 아이템을 사용해서 셀을 구성해주고 있습니다. 이 경우 기본 cell을 사용하기 때문에 prepareForReuse를 호출하지 않아도 재사용 이슈가 없을지, 그게 아니라 Custom Cell을 만들어서 재사용 이슈를 없애줘야 하는지 궁금합니다.

[de58a11](https://github.com/yagom-academy/ios-open-market/pull/184/commits/de58a11065cbf83226b628d1337a61cd331f8029)

Reply:
이 내용에 대해서 저희 팀에서 연구해본적이 있는데요. 결론은 어지간한 reset으로는 '큰 영향 없다' 는 것 이었습니다.
애플의 말은 prepareForReuse 시점에 UI 업데이트를 해서 main thread 부하를 주지 말라는 것인데요 . 만일 부하가 누적된다면 스크롤이 끊길 수 있기 때문입니다.

그런데 우리의 상황인 image = nil 정도로는 main thread에 부하를 주지 못합니다. AutoLayout 업데이트와 같은 것을 하지 말라는 것으로 이해해주시면 좋겠습니다 .

기본 Cell과 관련해서는 상황 마다 다르겠습니다만. 대부분의 재사용 관련 문제가 발생하는 이미지의 경우, cell configure시점에 API 통신 (서버에서 이미지를 가져온다던지) 을 해야한다면, reset이 필요해요~